src/wp-admin/includes/privacy-tools.php

```diff
@@ -388,14 +385,32 @@ function wp_privacy_generate_personal_data_export_file( $request_id ) {
 		),
 	);
 
-	// Merge in the special about group.
-	$groups = array_merge( array( 'about' => $about_group ), $groups );
+	// And now, all the Groups.
+	$groups = get_post_meta( $request_id, '_export_data_grouped', true );
 
-	$groups_count = count( $groups );
+	// Merge in the special about group.
+	if ( is_array( $groups ) ) {
+		$groups       = array_merge( array( 'about' => $about_group ), $groups );
+		$groups_count = count( $groups );
+	} elseif ( '' === $groups || false === $groups ) {
+		$groups       = array( 'about' => $about_group );
+		$groups_count = 1;
+	} else {
+		wp_send_json_error(
+			sprintf(
+				'Warning: array_merge(): Expected parameter 2 to be an array, %1$s given.',
+				gettype( $groups )
+			)
+		);
```

Comment:
Outline of the code to change, though the doing it wrong needs attention.

```suggestion
	// False will be returned when the request ID is invalid.
	if ( false === $groups ) {
		$groups       = array( 'about' => $about_group );
		$groups_count = 1;
	} else {
		if ( ! is_array ( $groups ) ) {
			// @TODO: Update this message.
			__doing_it_wrong(
				'',
				sprintf(
					'Warning: array_merge(): Expected parameter 2 to be an array, %1$s given.',
					gettype( $groups )
				)
			);
		}
		$groups            = array_merge( array( 'about' => $about_group ), (array) $groups );
		$groups_count = count( $groups );
	}			
```